tests/unit/s3/test_transfer.py

```diff
@@ -160,6 +240,25 @@ def test_download_file(self):
             'bucket', 'key', '/tmp/smallfile', extra_args, None
         )
 
+    def test_download_file_via_path(self):
+        extra_args = {
+            'SSECustomerKey': 'foo',
+            'SSECustomerAlgorithm': 'AES256',
+        }
+        self.transfer.download_file(
+            'bucket',
+            'key',
+            pathlib.Path('/tmp/smallfile'),
+            extra_args=extra_args,
+        )
+        self.manager.download.assert_called_with(
+            'bucket',
+            'key',
+            os.path.normpath('/tmp/smallfile'),
```

Comment:
Why is `os.path.normpath` needed here?

Reply:
I just tested it now, but the keys can be messed up on Windows:
```
Expected: download('bucket', 'key', '/tmp/smallfile', {'SSECustomerKey': 'foo', 'SSECustomerAlgorithm': 'AES256'}, None)
E           Actual: download('bucket', 'key', '\\tmp\\smallfile'
```

Reply:
> If you want to use an absolute path, make sure the first character is not a slash ('/') Use [pathlib](https://docs.python.org/3/library/pathlib.html#module-pathlib) or the [os.path.normpath()](https://docs.python.org/3/library/os.path.html#os.path.normpath) function to ensure that paths are properly encoded for Win32.

I will add `normpath` to ensure it is properly encoded independent of the system.